crates/ty_python_semantic/src/types/class.rs

```diff
@@ -243,11 +243,22 @@ impl CodeGeneratorKind {
 /// # Ordering
 /// Ordering is based on the generic aliases's salsa-assigned id and not on its values.
 /// The id may change between runs, or when the alias was garbage collected and recreated.
-#[salsa::interned(debug, heap_size=ruff_memory_usage::heap_size)]
-#[derive(PartialOrd, Ord)]
-pub struct GenericAlias<'db> {
```

Comment:
Did you consider keeping this a struct, and making just the `origin` field be an enum? Even if we need to add additional variants in the future, I can't imagine a generic alias not having a specialization, and it looks like a lot of the copied logic below is actually shared.